### DISCUSSION TITLE
Is it possible to change selected control color within a ListBoxItem?

### DISCUSSION THREAD

**Discussion by JaoHundred** [AvaloniaUI/Avalonia#13389] [Q&A] [answered]
Hi, I'm taking a look at styling and saw how easy is to change the `ListBoxItem` color using the `selected` pseudo class. I was wondering if its possible to apply the color change in a control which is inside the ListBoxItem,

```
<Style Selector="ListBox.filters ListBoxItem:selected /template/ ContentPresenter">
  <Setter Property="Background" Value="Transparent"/>

  <Style Selector="^ Border">
    <Setter Property="Background" Value="Red"/>
  </Style>
</Style>
```
How can I get to change the color in this situation?

**Comment by timunie** [comment, marked as answer]
`<Style Selector="ListBox.filters ListBoxItem:selected Border">` this should work for all Borders inside ListBoxItem iirc 

check docs about Style Selectors for reference.

**Reply by JaoHundred** [reply]
Thanks timunie, I just recently found about the dev tools it will be a great help to assist the correct use of selectors.

**Reply by glowysourworm** [reply]
@timunie There's a problem with the item container. How do you select just the ListBoxItem's container? For example, this code will work for setting the Border (in my data template) for the ListBoxItem. But, there's a small margin around it (or padding from the ItemContainer):

				<Style Selector="ListBoxItem:selected Border">
					<Setter Property="Background" Value="#FFFFFFFF" />
				</Style>

How can I write a selector for the ListBoxItem, not it's border? (I already tried removing "Border")

**Reply by timunie** [reply]
See the original style as reference. If you use fluentTheme: https://github.com/AvaloniaUI/Avalonia/blob/33c58a965fea365eec6825acff6528527610bff2/src/Avalonia.Themes.Fluent/Controls/ListBoxItem.xaml#L52-L57
